testutil/integration/app.go

```diff
@@ -780,44 +836,167 @@ func (app *App) nextBlockUpdateCtx() {
 		Time:    header.Time,
 	}
 
-	newContext := app.BaseApp.NewUncachedContext(true, header).
+	// NB: Intentionally omitting the previous contexts EventManager.
+	// This ensures that each event is only observed for 1 block.
+	newContext := app.NewUncachedContext(true, header).
 		WithBlockHeader(header).
 		WithHeaderInfo(headerInfo).
-		WithEventManager(prevCtx.EventManager()).
 		// Pass the multi-store to the new context, otherwise the new context will
 		// create a new multi-store.
 		WithMultiStore(prevCtx.MultiStore())
 	*app.sdkCtx = newContext
 }
 
-// CreateMultiStore is a helper for setting up multiple stores for provided modules.
-func CreateMultiStore(keys map[string]*storetypes.KVStoreKey, logger log.Logger) storetypes.CommitMultiStore {
-	db := dbm.NewMemDB()
-	cms := store.NewCommitMultiStore(db, logger, metrics.NewNoOpMetrics())
+// setupDefaultActorsState uses the integration app keepers to stake "default"
+// on-chain actors for use in tests. In creates a service, and stakes a supplier
+// and application as well as funding the bank balance of the default supplier.
+//
+// TODO_IMPROVE: Eliminate usage of and remove this function in favor of
```

Comment:
Could you mark this as `TODO_TECHDEBT(@bryanchriswhite)`. I think your're one of the few people who will be able to do a good job at this.